Dockerfile

```diff
@@ -255,13 +265,16 @@ RUN if [[ ${INSTALL_MYSQL_CLIENT} != "true" ]]; then \
     if [[ ${INSTALL_FROM_PYPI} == "true" ]]; then \
         if [[ "${UPGRADE_TO_NEWER_DEPENDENCIES}" != "false" ]]; then \
             pip install --user "${AIRFLOW_INSTALLATION_METHOD}[${AIRFLOW_EXTRAS}]${AIRFLOW_INSTALL_VERSION}" \
+                ${EAGER_UPGRADE_ADDITIONAL_REQUIREMENTS} \
                 --upgrade --upgrade-strategy eager; \
             pip install --upgrade "pip==${AIRFLOW_PIP_VERSION}"; \
+            pip check || ${CONTINUE_ON_PIP_CHECK_FAILURE}; \
         else \
             pip install --upgrade --upgrade-strategy only-if-needed \
                 --user "${AIRFLOW_INSTALLATION_METHOD}[${AIRFLOW_EXTRAS}]${AIRFLOW_INSTALL_VERSION}" \
                 --constraint "${AIRFLOW_CONSTRAINTS_LOCATION}"; \
             pip install --upgrade "pip==${AIRFLOW_PIP_VERSION}"; \
+            pip check || ${CONTINUE_ON_PIP_CHECK_FAILURE}; \
```

Comment:
We fail by default when we build the image locally but we can also disable it and continue building the image if we want to inspect it.